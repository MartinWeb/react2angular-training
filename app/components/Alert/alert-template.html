<div class="alert {{$ctrl.type}}" ng-if="$ctrl.isVisible" ng-click="$ctrl.isVisible = !$ctrl.isVisible">
    <p>{{$ctrl.title}}</p>
    <ng-transclude></ng-transclude>
</div>