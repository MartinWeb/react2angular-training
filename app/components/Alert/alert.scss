.alert {
    padding: 5px 10px;
    margin: 10px 0;

    &:hover {
        cursor: pointer;
    }
}

.info {
    background-color: dodgerblue;
    color: white;
}

.warning {
    background-color: rgb(241, 90, 36);
    color: white;
}