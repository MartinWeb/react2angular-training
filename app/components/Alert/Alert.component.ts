import angular from "angular";
import template from './alert-template.html';
import './alert.scss';

export class AlertController implements angular.IController {
    public isVisible: boolean;

    public $onInit() {
        this.isVisible = true;
    }
}

export const AlertComponent: angular.IComponentOptions = {
    template,
    controller: AlertController,
    bindings: {
        title: '@',
        type: '@'
    },
    transclude: true,
};