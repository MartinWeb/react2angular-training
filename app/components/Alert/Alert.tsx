import { ReactNode, useState } from "react";
import './alert.scss';

export interface AlertProps {
  title: string;
  type: string;
  children?: ReactNode;
}

const Alert = ({ title, type, children }: AlertProps) => {
    const [isVisible, setIsVisible] = useState(true);

    return (
        <>
            {isVisible ? 
                <div className={`alert ${type}`}>
                    <h4>{title}</h4>
                    <span className="close" onClick={() => setIsVisible(false)}>X</span>
                    {children}
                </div> 
            : null}
        </>
    );
};

export { Alert };
