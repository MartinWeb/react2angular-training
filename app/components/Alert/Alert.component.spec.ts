import { AlertComponent, AlertController } from "./Alert.component";

describe('AlertComponent', () => {
    test('should have valid snapshot', () => {
        expect(AlertComponent.template).toMatchSnapshot();
    });

    it('should have correct bindings', () => {
        expect(AlertComponent.bindings).toMatchSnapshot();
    });
});

describe('AlertController', () => {
    let target: AlertController;

    beforeEach(() => {
        target = new AlertController();
    });

    it('$onInit', () => {
        target.$onInit();
        expect(target.isVisible).toEqual(true);
    });
});