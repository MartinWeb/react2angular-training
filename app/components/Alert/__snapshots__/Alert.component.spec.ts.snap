// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertComponent should have correct bindings 1`] = `
{
  "title": "@",
  "type": "@",
}
`;

exports[`AlertComponent should have valid snapshot 1`] = `
"<div class="alert {{$ctrl.type}}" ng-if="$ctrl.isVisible" ng-click="$ctrl.isVisible = !$ctrl.isVisible">
    <p>{{$ctrl.title}}</p>
    <ng-transclude></ng-transclude>
</div>"
`;
